docker-compose.yml

```diff
@@ -0,0 +1,211 @@
+version: '3.8'
+
+services:
+  angular-build:
+    build:
+      context: .
+      dockerfile: ./docker/Dockerfile
+    container_name: oppia-angular-build
+    volumes:
+      - .:/app/oppia
+      - shared_node_modules:/app/oppia/node_modules
+      - shared_third_party:/app/oppia/third_party
+      - shared_proto_files:/app/oppia/proto_files
+      - shared_js_ts_proto_files:/app/oppia/extensions/classifiers/proto
+      - shared_assets_dir:/app/oppia/assets
+      - shared_angular_build:/app/oppia/dist
+    healthcheck:
+      test: test "$(ls -A /app/oppia/dist)" || exit 1
+      timeout: 30s
+      interval: 30s
+      retries: 5
+      start_period: 20m
+    # added max-old-space-size to avoid out of memory error, check this link for more info
+    # https://stackoverflow.com/questions/62663167/dockerizing-react-in-production-mode-fatal-error-ineffective-mark-compacts-nea
+    command: >
+      sh -c '
+        trap "rm /app/oppia/node_modules/@angular/compiler-cli/ngcc/__ngcc_lock_file__" EXIT
+        export NODE_OPTIONS=--max_old_space_size=4096 && /app/oppia/node_modules/.bin/ng build --watch
```

Comment:
max__space_size is only required in prod mode right?

Reply:
umm, please refer here: https://stackoverflow.com/questions/62663167/dockerizing-react-in-production-mode-fatal-error-ineffective-mark-compacts-nea
also I have already added a comment above explaining this

Reply:
I'm not sure which answer to check, but it sounds like it's needed in prod env, this PR is intended to only run the dev env, so it it needed right now?

Reply:
yes, I am assiging a minimum amount of `max_old_space_size ` for the ng-build process so it wont face out of memory heap issue. if I remove this, it is failing with such error of out of memory

Reply:
Assign it through the environment key
```
environment:
   - NODE_OPTIONS=--max_old_space_size=4096
command: ...
```

Reply:
done + tested